From c9669f332fb3095c3eb0681034959287312f2c3c Mon Sep 17 00:00:00 2001
From: Chris PeBenito <Christopher.PeBenito@microsoft.com>
Date: Mon, 23 May 2022 15:23:27 +0000
Subject: [PATCH 37/37] container: Getattr generic device nodes.

There should be no device_t device nodes, but add access in case they
exist. Saw containerd fail to start containers if it couldn't stat() all
devices.

Signed-off-by: Chris PeBenito <Christopher.PeBenito@microsoft.com>
---
 policy/modules/services/container.te | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/policy/modules/services/container.te b/policy/modules/services/container.te
index 64b1ad918..709f2e214 100644
--- a/policy/modules/services/container.te
+++ b/policy/modules/services/container.te
@@ -388,6 +388,8 @@ corenet_rw_tun_tap_dev(container_engine_domain)
 
 dev_getattr_all_blk_files(container_engine_domain)
 dev_getattr_all_chr_files(container_engine_domain)
+dev_getattr_generic_blk_files(container_engine_domain)
+dev_getattr_generic_chr_files(container_engine_domain)
 dev_setattr_null_dev(container_engine_domain)
 dev_getattr_fs(container_engine_domain)
 dev_remount_fs(container_engine_domain)
-- 
2.25.1

